test/openssl/utils.rb

```diff
@@ -290,6 +290,14 @@ def check_component(base, test, keys)
       assert_equal base.send(comp), test.send(comp)
     }
   end
+
+  def assert_sign_verify_false_or_error
+    begin
+      assert_equal(false, yield)
+    rescue => e
+      assert_kind_of(OpenSSL::PKey::PKeyError, e)
+    end
+  end
```

Comment:
Sorry, my earlier suggestion wasn't right as the `rescue` clause would incorrectly catch an assertion failure from `assert_equal` (which utilizes exceptions). I think something like this should work:

```suggestion
  def assert_sign_verify_false_or_error
    ret = yield
  rescue => e
    assert_kind_of(OpenSSL::PKey::PKeyError, e)
  else
    assert_equal(false, ret)
  end
```